lib/src/main/java/com/otaliastudios/transcoder/internal/thumbnails/DefaultThumbnailsEngine.kt

```diff
@@ -240,6 +240,27 @@ class DefaultThumbnailsEngine(
         tracks.updateTracksInfo()
     }
 
+    override fun updateDataSources(dataSourcesNew: List<DataSource>) {
+        val currentVideoIds = dataSources.videoOrNull()?.map { it.mediaId() }
+        val newSourceIds = dataSourcesNew.map { it.mediaId() }.distinct()
+        val toAdd = newSourceIds - currentVideoIds
+        val toRemove = currentVideoIds?.minus(newSourceIds)
+        toAdd.forEach { id ->
+            val source = dataSourcesNew.first { it.mediaId() == id }
+            dataSources.addVideoDataSource(
```

Comment:
we should use the addDataSource and removeDataSuurce methods here

Reply:
updated, please check